bennu-portal/src/main/java/org/fenixedu/bennu/portal/servlet/PortalInitializer.java

```diff
@@ -55,13 +54,40 @@ public void contextInitialized(ServletContextEvent sce) {
         }
 
         sce.getServletContext().setAttribute("portal", new PortalBean(sce.getServletContext()));
+
+        migrateMenus();
+    }
+
+    @Atomic(mode = Atomic.TxMode.WRITE)
+    private void migrateMenus() {
+        /**
+         * This logic can be removed after there is a deploy in all instances and visible field in MenuItem is removed
+         * #qubIT-Omnis-6137
+         *
+         */
+        migrateMenuItemVisibility(PortalConfiguration.getInstance().getMenu());
```

Comment:
@pabrantes although i performed this migration locally and I still had 13 results when i ran the query

select * from MENU_ITEM WHERE VISIBILITY IS NULL AND VISIBLE IS NOT NULL\G

Some of them didn't have parents so probably they are roots for the reamining ones and the dfs couldn't visit them because of that. Will this create a problem?

Reply:
Well we need to understand if they are just another unreachable tree. If so we're safe, otherwise if they are reachable somehow the transactions will fail because the required consistency predicate will throw an exception. Ping me on monday so we can look into the data and see what are those items.